ai_agents/job_posting_agent/agents/pipeline.py

```
@@ -0,0 +1,103 @@
+    notify("Generating with DeepSeek V4 Flash (single request)...")
+    print(f"[Pipeline] Calling NVIDIA NIM for {company} / {role}", flush=True)
+
+    try:
+        response = chat_completion(
+            [{"role": "user", "content": _build_prompt(company, role)}],
+            max_tokens=MAX_OUTPUT_TOKENS,
+        )
+    except (NvidiaAPIError, TimeoutError, ValueError, httpx.HTTPError) as exc:
+        raise ValueError(
+            f"Could not complete NVIDIA API request: {exc}\n\n"
+            "Wait 1–2 minutes and try again. Confirm NVIDIA_API_KEY in .env "
```

Ruff (0.15.12)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)